security/sshdo/files/macports.patch

```diff
@@ -0,0 +1,131 @@
++	[ -d $(MANDIR)/man8 ] || mkdir -p -m 755 $(MANDIR)/man8
++	install -o root -g wheel -m 644 sshdo.8.gz $(MANDIR)/man8
++	[ -d $(MANDIR)/man5 ] || mkdir -p -m 755 $(MANDIR)/man5
++	install -o root -g wheel -m 644 sshdoers.5.gz $(MANDIR)/man5
+ 
+ uninstall:
+ 	rm -f $(BINDIR)/$(BIN)
+--- sshdo.orig	2021-09-08 21:02:40.000000000 +1000
++++ sshdo	2023-06-12 18:31:28.000000000 +1000
+@@ -21,7 +21,7 @@
+ 
+ from __future__ import print_function
+ import os, sys, re, getopt, syslog, pwd, grp, glob, gzip, copy, operator, functools
+-py2 = sys.version_info.major == 2
++py2 = (sys.version_info[0] if type(sys.version_info) == tuple else sys.version_info.major) == 2
+ none = None
+ 
+ prog_name = 'sshdo'
+@@ -30,8 +30,8 @@
+ prog_author = 'raf <raf@raf.org>'
+ prog_url = 'https://raf.org/sshdo'
+ 
+-default_config_fname = '/etc/sshdoers'
+-default_logfiles = '/var/log/auth.log*'
++default_config_fname = '/opt/local/etc/sshdoers'
++default_logfiles = '/var/log/system.log*'
+ default_match = 'digits'
+ syslog_facility_map = dict(auth=syslog.LOG_AUTH, daemon=syslog.LOG_DAEMON, user=syslog.LOG_USER, local0=syslog.LOG_LOCAL0, local1=syslog.LOG_LOCAL1, local2=syslog.LOG_LOCAL2, local3=syslog.LOG_LOCAL3, local4=syslog.LOG_LOCAL4, local5=syslog.LOG_LOCAL5, local6=syslog.LOG_LOCAL6, local7=syslog.LOG_LOCAL7)
+ 
+--- sshdoers.orig	2023-06-11 22:37:00.000000000 +1000
++++ sshdoers	2023-06-12 15:43:31.000000000 +1000
+@@ -1,7 +1,7 @@
+-# /etc/sshdoers: Configure sshdo and specify the commands that
++# /opt/local/etc/sshdoers: Configure sshdo and specify the commands that
+ # are allowed via incoming ssh connections where authorized
+-# keys include command="/usr/bin/sshdo" or where a ForceCommand
+-# directive in /etc/ssh/sshd_config specifies /usr/bin/sshdo.
++# keys include command="/opt/local/bin/sshdo" or where a ForceCommand
++# directive in /etc/ssh/sshd_config specifies /opt/local/bin/sshdo.
+ #
+ # See the manual pages sshdo(8) and sshdoers(5) for details.
+ 
+@@ -23,13 +23,13 @@
+ # Change the default log files glob pattern(s) for the --learn
+ # and --unlearn options, if necessary.
+ #
+-# logfiles /var/log/auth.log*
++# logfiles /var/log/system.log*
+ 
+ # Uncomment the banner directive and modify the banner file, if
+ # necessary, to output its contents to standard error (stderr)
+ # for ssh clients whose commands are disallowed and not executed.
+ #
+-# banner /etc/sshdo.banner
++# banner /opt/local/etc/sshdo.banner
+ 
+ # Change which characters are matched by hash characters ("#")
+ # in shell commands in authorization directives, if necessary.
+@@ -45,6 +45,6 @@
+ #
+ # training
+ 
+-# Add command authorization directives here or in /etc/sshdoers.d/*
++# Add command authorization directives here or in /opt/local/etc/sshdoers.d/*
+ # (e.g. "user1 user2/label +group3 -user3: cmd arg").
+ 
+--- test_sshdo.orig	2021-09-08 21:02:25.000000000 +1000
++++ test_sshdo	2023-06-12 16:08:39.000000000 +1000
```

Comment:
I wonder how much of this patch should actually be part of the upstream source?

Reply:
Probably most of it. I can add a configure script for the next version and make the patch unnecessary or much smaller, but it's what's needed for v1.1.

Reply:
I've upgraded it to v1.1.1 which has a configure script that eliminates the patch that was needed for v1.1.